app/service/search_index.rb

```diff
@@ -1,60 +1,35 @@
 class SearchIndex
-  def initialize(model_name:, id:, async: false)
-    @model_name = model_name
-    @id = id
-    @async = async
-  end
-
-  def add
-    if async
-      add_to_index_asynchronously
-    else
-      record = model_name.constantize.find(id)
-      record.__elasticsearch__.index_document
-    end
+  def self.search_index_callbacks_enabled?
+    ENV.fetch('ENABLE_SEARCH_INDEX_CALLBACKS', true) != 'false'
   end
 
-  def remove
-    if async
-      remove_from_index_asynchronously
-    else
-      Elasticsearch::Model.client.delete(
-        index: model_name.constantize.index_name,
-        type: model_name.downcase,
-        id: id,
+  def self.log_callback_warning
+    Rails.logger.tagged('ELASTICSEARCH') do
```

Comment:
I notice these three same lines are repeated 3 times with only the warning content changing. Maybe it would be nice to have a single method that deals with logging to which we could pass the line to log? Maybe a class method on `SearchIndex` or a very small and simple ElasticSearch logger class where we could easily add more logging options later on (if we need it)?

A quick example for reference:

```ruby
def self.log_es_warning(warning)
  Rails.logger.tagged('ELASTICSEARCH') do
    Rails.logger.warn(warning)
  end
end
```